app/bundles/NotificationBundle/Helper/NotificationUploader.php

```diff
@@ -0,0 +1,145 @@
+use Symfony\Component\Form\Form;
+
+class NotificationUploader
+{
+    /**
+     * @var FileUploader
+     */
+    private $fileUploader;
+
+    /**
+     * @var CoreParametersHelper
+     */
+    private $coreParametersHelper;
+
+    /**
+     * @var array
+     */
+    private $uploadFilesName = ['actionButtonIcon1', 'actionButtonIcon2', 'icon', 'image'];
+
+    public function __construct(FileUploader $fileUploader, CoreParametersHelper $coreParametersHelper)
+    {
+        $this->fileUploader         = $fileUploader;
+        $this->coreParametersHelper = $coreParametersHelper;
+    }
+
+    /**
+     * @param Notification $notification
+     * @param $request
+     * @param Form $form
+     */
+    public function uploadFiles(Notification $notification, $request, Form $form)
+    {
+        $files         = $request->files->all()['notification'];
+
+        $uploadedFiles = [];
+        $deteledFiles  = [];
+        foreach ($this->getUploadFilesName() as $fileName) {
+            $uploadDir    = $this->getUploadDir($notification);
+
+            // Delete file
+            if (!empty($form->get($fileName.'_delete')->getData())) {
+                $this->fileUploader->delete($uploadDir.DIRECTORY_SEPARATOR.$this->getEntityVar($notification, $fileName));
+                $this->getEntityVar($notification, $fileName, 'set', '');
+            }
+
+            /* @var UploadedFile $file */
+            if (empty($files[$fileName])) {
+                continue;
+            }
+            $file = $files[$fileName];
+
+            try {
+                $uploadedFile = $this->fileUploader->upload($uploadDir, $file);
+                $this->getEntityVar($notification, $fileName, 'set', $uploadedFile);
+            } catch (FileUploadException $e) {
+                $this->fileUploader->delete($uploadDir.DIRECTORY_SEPARATOR.$uploadedFile);
+            }
+        }
+    }
+
+    /**
+     * @param Notification $notification
+     * @param string       $fileName
+     *
+     * @return string
+     */
+    public function getCompleteFilePath(Notification $notification, $fileName)
+    {
+        $uploadDir = $this->getUploadDir($notification);
+
+        return $uploadDir.DIRECTORY_SEPARATOR.$fileName;
+    }
+
+    /**
+     * @param Notification $notification
+     */
+    public function deleteAllFilesOfNotification(Notification $notification)
+    {
+        $uploadDir = $this->getUploadDir($notification);
+        $this->fileUploader->delete($uploadDir);
+    }
+
+    /**
+     * @param $notification
+     *
+     * @return string
+     */
+    public function getFullUrl($notification, $key)
+    {
+        if ($fileName = $this->getEntityVar($notification, $key)) {
+            return $this->coreParametersHelper->getParameter('site_url').DIRECTORY_SEPARATOR.$this->coreParametersHelper->getParameter('notification_upload_dir').DIRECTORY_SEPARATOR.$notification->getId().DIRECTORY_SEPARATOR.$fileName;
```

Comment:
This won't work in all setups. We'll need a different parameter for the URL portion mainly because of the note below.

Reply:
If using the PathsHelper and image_path as per below, then use the AssetsHelper::getUrl (ensure $absolute = true and $ignorePrefix = true) instead of concatenating.